src/components/structures/RoomView.js

```diff
@@ -504,17 +526,21 @@ module.exports = React.createClass({
         }
     },
 
-    _paginateCompleted: function() {
-        debuglog("paginate complete");
-
-        // we might have switched rooms since the paginate started - just bin
+    _onTimelineUpdated: function(gotResults) {
+        // we might have switched rooms since the load started - just bin
         // the results if so.
         if (this.unmounted) return;
 
         this.setState({
-            room: MatrixClientPeg.get().getRoom(this.props.roomId),
             paginating: false,
         });
+
+        if (gotResults) {
+            this.setState({
+                events: this._timelineWindow.getEvents(),
```

Comment:
Okay. We'll see how it goes in the long run for maintainability. Slightly nasty that the getter is a costly operation though; it's not clear/hinted at anywhere.